### ISSUE TITLE
Unable to use class decorator with static member functions

### ISSUE THREAD

**Issue by ahagelstein** [Microsoft/TypeScript#11322]
**TypeScript Version:**  typescript@2.1.0-dev.20161102

**Code**

```
@testDecorator()
export class TestClass {

    public static create() {

        return new TestClass();
    }
}
```

**Expected behavior:**

typescript@2.0.6 generates the correct code. Here the static function "create" uses the decorated constructor function (the variable declared in the first line). 

```
var TestClass = (function () {
    function TestClass() {
    }
    TestClass.create = function () {
        return new TestClass();
    };
    TestClass = __decorate([            <-- correct position
        testDecorator()
    ], TestClass);
    return TestClass;
}());
exports.TestClass = TestClass;
```

**Actual behavior:** 

typescript@2.1.0-dev.201620161102 generates the wrong code. Here the static function "create" uses the variable declared in the first line, but this is not the decorated constructor function. 

```
var TestClass = (function () {
    function TestClass() {
    }
    TestClass.create = function () {
        return new TestClass();
    };
    return TestClass;
}());
TestClass = __decorate([           <-- wrong position
    testDecorator()
], TestClass);
exports.TestClass = TestClass;
```

For a detailed code sample look at #10840.

**Comment by ahagelstein** [comment]
@rbuckton : From my point of view this bug still exists in the latest nightly. Could you please give me a feedback why the position of the decorator code was changed or is there a plan to fix it?

**Comment by rbuckton** [comment]
@ahagelstein: If, for example, you wanted to have a decorator that set `writable` to `false` to make a static field read-only, this would not work if static assignments occurred *after* decorators were run. We will not be changing the emit for decorators until the official proposal for decorators for ECMA-262 advances, as it is a significant change from the current approach.

**Comment by ahagelstein** [comment]
What I need for my implementation is a decorator which monitors the constructor function of the decorated class (see #10840). If an instance is created the decorator code is called because the decorator creates a new construtor function for the class (a kind of inheritance).
With typescript@2.0.6 there is no problem.
But with typescript@2.1.0-dev.201620161102 the code of the decorator has moved (see example above) and the decorator code is no longer visible from a static method within the class. If I create the instance outside the decorated class the code works.
Is the NEW position of the decorator code in typescript@2.1.0-dev.201620161102 the final position?

**Comment by ahagelstein** [comment]
Closed with #12247. Thanks.
